### ISSUE TITLE
vim 9.0.1750 bottle on linux seems to depend on python 3.11.4_1 while current python 3.11 is 3.11.5

### ISSUE THREAD

**Issue by huyz** [Homebrew/homebrew-core#140578]
### `brew gist-logs <formula>` link OR `brew config` AND `brew doctor` output

```shell
❯ brew gist-logs vim
Error: No logs.

❯ brew config
HOMEBREW_VERSION: 4.1.6
ORIGIN: https://github.com/Homebrew/brew
HEAD: 3cd72905ce3d36a84994cc6c6b9ebbdafff4e922
Last commit: 8 days ago
Core tap JSON: 26 Aug 21:31 UTC
HOMEBREW_PREFIX: /home/linuxbrew/.linuxbrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_EDITOR: vim
HOMEBREW_MAKE_JOBS: 1
Homebrew Ruby: 2.6.10 => /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.10_1/bin/ruby
CPU: single-core 64-bit skylake
Clang: N/A
Git: 2.34.1 => /bin/git
Curl: 7.81.0 => /bin/curl
Kernel: Linux 5.15.0-79-generic x86_64 GNU/Linux
OS: Ubuntu 22.04.3 LTS (jammy)
Host glibc: 2.35
/usr/bin/gcc: 11.4.0
/usr/bin/ruby: 3.0.2
glibc: N/A
gcc@11: N/A
gcc: N/A
xorg: N/A

❯ brew doctor
Your system is ready to brew.
```

### Verification

- [X] My "`brew doctor` output" says `Your system is ready to brew.` and am still able to reproduce my issue.
- [X] I ran `brew update` and am still able to reproduce my issue.
- [X] I have resolved all warnings from `brew doctor` and that did not fix my problem.
- [X] I searched for recent similar issues at https://github.com/Homebrew/homebrew-core/issues?q=is%3Aissue and found no duplicates.

### What were you trying to do (and why)?

I'm trying to use python3 functionality inside vim.

### What happened (include all command output)?

When I run `vim -u NONE -U NONE -N -i NONE`, then run the command `:python3 print(sys.version)`, vim crashes and messes up my terminal.

The output looks like this:
```
~
~
~
~
~
~
Could not find platform independent libraries <prefix>
                                                      Could not find platform dependent libraries <exec_prefix>
                                                                                                               Python path configuration:
                 PYTHONHOME = (not set)
                                         PYTHONPATH = (not set)
                                                                 program name = 'python3'
                                                                                           isolated = 0
                                                                                                         environment = 1
                                                                                                                          user site = 1
               safe_path = 0
                              import site = 1
                                               is in build tree = 0
                                                                     stdlib dir = '/home/linuxbrew/.linuxbrew/Cellar/python@3.11/3.11.4_1/lib/python3.11'
                                 sys._base_executable = '/home/vuul/.pyenv/shims/python3'
                                                                                           sys.base_prefix = '/home/linuxbrew/.linuxbrew/Cellar/python@3.11/3.11.4_1'
                                             sys.base_exec_prefix = '/home/linuxbrew/.linuxbrew/Cellar/python@3.11/3.11.4_1'
    sys.platlibdir = 'lib'
                            sys.executable = '/home/vuul/.pyenv/shims/python3'
                                                                                sys.prefix = '/home/linuxbrew/.linuxbrew/Cellar/python@3.11/3.11.4_1'
                             sys.exec_prefix = '/home/linuxbrew/.linuxbrew/Cellar/python@3.11/3.11.4_1'
                                                                                                         sys.path = [
                                                                                                                         '/home/linuxbrew/.linuxbrew/Cellar/python@3.11/3.11.4_1/lib/python311.zip',
                                                                              '/home/linuxbrew/.linuxbrew/Cellar/python@3.11/3.11.4_1/lib/python3.11',
                                '/home/linuxbrew/.linuxbrew/Cellar/python@3.11/3.11.4_1/lib/python3.11/lib-dynload',
                                                                                                                      ]
                                                                                                                       Fatal Python error: init_fs_encoding: failed to get the Python codec of the filesystem encoding
                                                                                            Python runtime state: core initialized
        ModuleNotFoundError: No module named 'encodings'

                                                        Current thread 0x00007f3cfe07fe40 (most recent call first):
                                                                                                                     <no Python frame>
            %
```

### What did you expect to happen?

I expected no crash and the output to look like vim on macOS:
```
3.11.5 (main, Aug 24 2023, 15:09:45) [Clang 14.0.3 (clang-1403.0.22.14.1)]
```

### Step-by-step reproduction instructions (by running `brew` commands)

```shell
1. `brew install vim`
```

**Comment by SMillerDev** [comment]
What does `brew info vim` say?

**Comment by huyz** [comment]
```shell
❯ brew info vim
==> vim: stable 9.0.1750 (bottled), HEAD
Vi 'workalike' with many additional features
https://www.vim.org/
Conflicts with:
  ex-vi (because vim and ex-vi both install bin/ex and bin/view)
  macvim (because vim and macvim both install vi* binaries)
/home/linuxbrew/.linuxbrew/Cellar/vim/9.0.1750 (2,170 files, 46.7MB) *
  Poured from bottle using the formulae.brew.sh API on 2023-08-26 at 14:32:06
From: https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/v/vim.rb
License: Vim
==> Dependencies
Required: gettext ✔, libsodium ✔, lua ✔, ncurses ✔, perl ✔, python@3.11 ✔, ruby ✔
==> Options
--HEAD
        Install HEAD version
==> Analytics
install: 25,802 (30 days), 65,798 (90 days), 124,030 (365 days)
install-on-request: 25,519 (30 days), 65,139 (90 days), 123,163 (365 days)
build-error: 25 (30 days)
````

**Comment by branchvincent** [comment]
I see you're using pyenv, does `pyenv rehash` help?

**Comment by huyz** [comment]
That doesn't help ❌ and neither does `pyenv global system` ❌.

But disabling `pyenv` altogether worked and prints out `3.11.4 (main, Jun  6 2023, 22:16:46) [GCC 11.3.0]`  ✅.
Hmm, I didn't realize I had 3.11.4 anywhere on my system. 🤷 

Strange, as I get the same system python for (a) `pyenv global system` and (b) disabling pyenv:
```shell
❯ python --version
Python 3.10.12
```
which is `/usr/bin/python`. 🤷 

Also, when I once again try with pyenv and vim, but this time using **system vim 8.2.213** (Ubuntu 22.04), then it works and prints out `3.10.12 (main, Jun 11 2023, 05:26:28) [GCC 11.4.0]` ✅, even if I change to `pyenv global 3.11.5` ✅.

To recap:
- `pyenv global system` vs. `pyenv global 3.11.5` seems to make no difference whether for Homebrew vim or system vim
- Having pyenv enabled breaks Homebrew vim ❌ but not system vim ✅.

**Comment by chenrui333** [comment]
Since this is more pf python path issue, I am gonna close it now.
